apps/etterlatte-behandling/src/main/kotlin/behandling/Behandling.kt

```diff
@@ -68,17 +67,25 @@ data class Foerstegangsbehandling(
     override val type: BehandlingType = BehandlingType.FØRSTEGANGSBEHANDLING,
     override val persongalleri: Persongalleri,
     override val kommerBarnetTilgode: KommerBarnetTilgode?,
-    private var virkningstidspunkt: Virkningstidspunkt?,
+    val virkningstidspunkt: Virkningstidspunkt?,
     val soeknadMottattDato: LocalDateTime,
     val gyldighetsproeving: GyldighetsResultat?
 ) : Behandling {
     fun hentVirkningstidspunkt() = virkningstidspunkt
-    fun oppdaterVirkningstidspunkt(dato: YearMonth, kilde: Grunnlagsopplysning.Saksbehandler) {
-        if (BehandlingStatus.kanRedigeres().contains(this.status)) {
-            virkningstidspunkt = Virkningstidspunkt(dato, kilde)
-        } else {
+    fun oppdaterVirkningstidspunkt(dato: YearMonth, kilde: Grunnlagsopplysning.Saksbehandler): Foerstegangsbehandling {
+        if (!BehandlingStatus.kanRedigeres().contains(this.status)) {
             throw RuntimeException("Kan ikke endre virkningstidspunkt for behandling som ikke er under behandling.")
         }
+
+        return this.copy(virkningstidspunkt = Virkningstidspunkt(dato, kilde))
+    }
+
+    fun oppdaterKommerBarnetTilgode(kommerBarnetTilgode: KommerBarnetTilgode): Foerstegangsbehandling {
```

Comment:
Burde det være en generell regel siden det treffer ting likt(at en behandling kan redigeres)?
Skal vi heller bare sjekke `behandling. kanRedigeres` ?

Reply:
```
if (BehandlingStatus.kanRedigeres().contains(this.status)) {
return this.copy(kommerBarnetTilgode = kommerBarnetTilgode)
        }
            throw RuntimeException("Kan ikke endre kommer barnet til gode for behandling som ikke er under behandling")
        
```

Reply:
Er det ikke litt rart at ikke `this.status` ikke kan kalle på `kanRedigeres` selv?

Reply:
Feks skrive om til `fun kanRedigeres(status: BehandlingStatus) = (underBehandling() - FATTET_VEDTAK).contains(status)` i enum klassen `BehandlingStatus` ?

Reply:
Godt inspill, skal få sett på dette.